tree/tree/src/TLeafD32.cxx

```diff
@@ -48,9 +48,13 @@ TLeafD32::TLeafD32(TBranch *parent, const char *name, const char *type) : TLeaf(
    fValue = nullptr;
    fPointer = nullptr;
    fElement = nullptr;
-
-   if (strchr(type, '['))
+   
+   if (strchr(type, '[')) {
```

Comment:
Should this depend on the value of `fTitle` instead? (which should have set by the base class constructor).